packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -87,13 +88,46 @@ class _HeroTag {
 Widget _wrapWithBackground({
   Border border,
   Color backgroundColor,
+  BoxDecoration background,
   Brightness brightness,
   Widget child,
   bool updateSystemUiOverlay = true,
 }) {
   Widget result = child;
   if (updateSystemUiOverlay) {
-    final bool isDark = backgroundColor.computeLuminance() < 0.179;
+    bool computeLuminance() {
+      double luminanceThreshold = 0.179;
+      // If the background contains a gradient
+      // calculate the average luminance for the array of colors
+      // and use it to determine wether it is considered
+      // a [darkbackground]
+      //
+      // First and last colors are given a stronger value
+      // and account for twice their real value
+      // as those are closer to the icons in the status bar
+      //
+      // If the color property of the background is set
+      // it will override the gradient's computed luminance
+      // despite the color itself being ignored
+      // This gives the developer the ability to manually
+      // change the luminance if so desired
+      if (background?.color != null) {
+        return background.color.computeLuminance() < luminanceThreshold;
+      } else if (background?.gradient != null) {
+        double gradientLuminanceThreshold = 0.0;
+        gradientLuminanceThreshold +=
```

Comment:
why do you need to special case the first and last ones? Just the for loop should be fine no?